test/MockSmartStore.js

```diff
@@ -314,7 +314,13 @@ var MockSmartStore = (function(window) {
                     name: "Count of soup items", 
                     example: "SELECT count(*) FROM {soupName}", 
                     pattern: /SELECT count\(\*\) FROM {(.*)}/i, 
-                    processor: this._smartQuerySoupCount 
+                    processor: this._smartQuerySoupCount
+                },
+                { 
+                    name: "Comparing soup item to integer with optional order by", 
+                    example: "SELECT {soupName:_soup} FROM {soupName} WHERE {soupName:whereField} > 123456 ORDER BY LOWER({soupName:orderByField})", 
+                    pattern: /SELECT {(.*):_soup} FROM {(.*)} WHERE {(.*):(.*)} ([!=<>]+) ([0-9]+)(?: ORDER BY LOWER\({(.*):(.*)}\))?/i,
```

Comment:
Supporting more comparisons types than the original pull request
Keeping the regexp for the comparator simple - the code will only accept !=,=,<,<=,>=